### DISCUSSION TITLE
Quasar2 cli(with vite) + alias for external dir + quasar build -> *.vue can't load.

### DISCUSSION THREAD

**Discussion by 1tkd** [quasarframework/quasar#15640] [General - Components / Directives / etc]
**[Brief explanation]**
In the main-code of the project below
https://github.com/1tkd/quasarMonorepoNG
Code that works fine with quasar dev doesn't work with quasar build + quasar serve.

**[Detailed explanation]**
1. In order to manage utility module and Vue components commonly used by multiple Quasar projects in a single library project,I defined   aliases from the reference Quasar projects to the library project as follows.

main-code/tsconfig.json
```
{
  "extends": "@quasar/app-vite/tsconfig-preset",
  "compilerOptions": {
    "baseUrl": ".",
+  "paths":{
+     "lib/*":["../lib-code/src/*"],
+     "src/*": ["src/*"],
+ ...
+     "stores/*": ["src/stores/*"]
    }
  },
}
```

main-code/quasar.config.js
```
    build: {
      target: {
        browser: [ 'es2019', 'edge88', 'firefox78', 'chrome87', 'safari13.1' ],
        node: 'node16'
      },

+      alias:{
+       'lib': path.resolve(__dirname, '../lib-code/src') 
+     },
```

2. Create utility module and vue component in lib-code.
+ lib-code/src/components/LibComp.vue
+ lib-code/src/components/utilcode.ts

3. import & use lib-code from main-code

main-code/src/components)/ExampleComponent.vue
```
<template>
  <div>
    <p>mainComp {{ func1() }}</p>
    <LIbComp></LIbComp>
  </div>
</template>

<script setup lang="ts">
import LIbComp from 'lib/components/LibComp.vue'
import {func1} from 'lib/components/utilcode'
</script>
```

5. run quasar dev => LibComp loaded correctly.

6. run quasar build & quasar serve dist/spa => LibComp failed to load."

**As a supplement,**
the following options were not chosen due to their respective reasons.
- App Extensions: There are constraints on ext-id and insufficient information for this usage.
- UI Kit: It seems to target only one component per project.
- npm: The library project undergoes frequent changes, and we do not want to go through a build process every time we make changes to the library.

**Comment by suleimanas** [comment]
check out rollupOptions > external

**Reply by 1tkd** [reply]
Thank you for your response. However, I'm sorry to say that I didn't understand the advice you gave me.

Are you suggesting that I should get Quasar or Vue as an external using the UMD method? I feel that this would result in a large disadvantage as it would prevent tree shaking.

Or are you advising that I should specify lib-code as an external? In that case, I would need to build lib-code itself as a library, and it would not be included in tree shaking.

All I want to achieve is to share the same file across multiple Quasar projects.
As an example, I am hoping for a behavior similar to development using symbolic links. (However, I have confirmed that using symbolic links causes LibComp to stop working with quasar build.)

**Comment by suleimanas** [comment]
how about extending another tsconfig, I haven't tried it yet but let's say your tsconfig in the project extends another common tsconfig that have the libComp in it's root

**Reply by 1tkd** [reply]
Thank you for your help.

I may be making inaccurate verifications as I am not familiar with the specifications of vite and tsconfig and do not fully understand the intention behind them. However, I have confirmed that LibComp is not displayed by trying the following two methods:

A. Recreating lib-code using npm init quasar and building without changing the Alias configuration for LibComp.
B.  After verifying A, placing LibComp at the same level as tsconfig (=baseUri) and rewriting Alias and import paths.

https://github.com/1tkd/quasarMonorepoExtTsconfig
